v2/split_slice.go

```diff
@@ -0,0 +1,29 @@
+package pie
+
+// SplitSlice returns splited slices which length equals splitLength.
+//
+// for this [1,2,3] slice with splitLength == 2 will return [][]int{{1,2}, {3}}.
+// If input slice length is less than splitLength, for this [1,2,3] slice with splitLength == 4
+// 		will return [][]int{{1,2,3}}.
+// If input slice length == 0 or splitLength <= 0, will return [][]T{}.
+func SplitSlice[T any](ss []T, splitLength int) [][]T {
+	result := make([][]T, 0)
+	l := len(ss)
+	if l == 0 || splitLength <= 0 {
+		return result
```

Comment:
I think this should panic if the splitLength is 0 or less. These are ambiguous values that might heavily hint towards a bug in the callers code.

Reply:
I think a panic typically means something went unexpectedly wrong. Mostly we use it to fail fast on errors that shouldn’t occur during normal operation and we aren’t prepared to handle gracefully.  I think it's elegant to return the default value here. After all, panic will terminate the process.
In our company, we only allow the use of panic when the server is in the start-up phase, if it has been successfully started, all errors should be handled with go error, not panic, except for system panic thrown out go sdk.
There are also some open source projects that do this, such as [cast](https://github.com/spf13/cast), their slogan "Don’t Panic! ... Cast"

Reply:
done